File: output/java/Medium/640-maximum-subarray.java
/**https://leetcode.com/problems/maximum-subarray/ */
//Given an integer array nums, find the subarray with the largest sum, and return its sum.


class MaximumSubarray {
    public int maxSubArray(int[] nums) {
        int max = nums[0];
        int sum = 0;
        for(int i = 0; i < nums.length; i++){
            sum += nums[i];
            max = Math.max(max, sum);
            if(sum < 0){
                sum = 0;
            }
        }
        return max;
        
        
    }
}

    

File: output/py/Medium/640-maximum-subarray.py
#https://leetcode.com/problems/maximum-subarray/
#Given an integer array nums, find the subarray with the largest sum, and return its sum.


class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """ 
        max_sum = nums[0]
        for i in range(1, len(nums)):
            if nums[i - 1] > 0:
                nums[i] += nums[i - 1]
            max_sum = max(max_sum, nums[i])
        return max_sum
        

    